api/broker/helpers.go

```diff
@@ -0,0 +1,168 @@
+	serviceTagsBytes, err := json.Marshal(service.Tags)
+	if err != nil {
+		return fmt.Errorf("could not marshal service tags: %s", err)
+	}
+	serviceRequiresBytes, err := json.Marshal(service.Requires)
+	if err != nil {
+		return fmt.Errorf("could not marshal service requires: %s", err)
+	}
+	serviceMetadataBytes, err := json.Marshal(service.Metadata)
+	if err != nil {
+		return fmt.Errorf("could not marshal service metadata: %s", err)
+	}
+
+	serviceOffering.Name = service.Name
+	serviceOffering.Description = service.Description
+	serviceOffering.Bindable = service.Bindable
+	serviceOffering.InstancesRetrievable = service.BindingsRetrievable
+	serviceOffering.BindingsRetrievable = service.BindingsRetrievable
+	serviceOffering.PlanUpdatable = boolPointerToBool(service.PlanUpdatable, serviceOffering.PlanUpdatable)
+	serviceOffering.CatalogID = service.ID
+	serviceOffering.CatalogName = service.Name
+	serviceOffering.Tags = json.RawMessage(serviceTagsBytes)
+	serviceOffering.Requires = json.RawMessage(serviceRequiresBytes)
+	serviceOffering.Metadata = json.RawMessage(serviceMetadataBytes)
+
+	return nil
+}
+
+func osbcCatalogPlanToServicePlan(servicePlan *types.ServicePlan, plan *catalogPlanWithServiceOfferingID) error {
+	planMetadataBytes, err := json.Marshal(plan.Metadata)
+	if err != nil {
+		return fmt.Errorf("could not marshal plan metadata: %s", err)
+	}
+	schemasBytes := make([]byte, 0)
+	if plan.Schemas != nil {
+		schemasBytes, err = json.Marshal(plan.Schemas)
+		if err != nil {
+			return fmt.Errorf("could not marshal plan service instance create schema: %s", err)
+		}
+	}
+
+	servicePlan.Name = plan.Plan.Name
+	servicePlan.Description = plan.Plan.Description
+	servicePlan.CatalogID = plan.Plan.ID
+	servicePlan.CatalogName = plan.Plan.Name
+	servicePlan.Free = boolPointerToBool(plan.Plan.Free, servicePlan.Free)
+	servicePlan.Bindable = boolPointerToBool(plan.Plan.Bindable, plan.ServiceOffering.Bindable)
+	servicePlan.PlanUpdatable = boolPointerToBool(&plan.ServiceOffering.PlanUpdatable, servicePlan.PlanUpdatable)
+	servicePlan.Metadata = json.RawMessage(planMetadataBytes)
+	servicePlan.Schemas = schemasBytes
+	servicePlan.ServiceOfferingID = plan.ServiceOffering.ID
+
+	return nil
+}
+
+func osbcClient(ctx context.Context, createFunc osbc.CreateFunc, broker *types.ServiceBroker) (osbc.Client, error) {
+	config := osbc.DefaultClientConfiguration()
+	config.Name = broker.Name
+	config.URL = broker.BrokerURL
+	config.AuthConfig = &osbc.AuthConfig{
+		BasicAuthConfig: &osbc.BasicAuthConfig{
+			Username: broker.Credentials.Basic.Username,
+			Password: broker.Credentials.Basic.Password,
+		},
+	}
+	log.C(ctx).Debug("Building OSB client for service broker with name: ", config.Name, " accessible at: ", config.URL)
+	return createFunc(config)
+}
+
+func transformBrokerCredentials(ctx context.Context, broker *types.ServiceBroker, transformationFunc func(context.Context, []byte) ([]byte, error)) error {
```

Comment:
This should be generalized for type Secured and used for all types that implement Secured